### PR TITLE
crypto/x509/by_store.c: Add check for OPENSSL_strdup

### PR DESCRIPTION
**Pull request by JiangJias**
As the potential failure of the OPENSSL_strdup(),
it should be better to check the return value and
return error if fails.

Signed-off-by: Jiasheng Jiang <jiasheng@iscas.ac.cn>

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/x509/by_store.c [hunk 1]
```diff
@@ -109,6 +109,8 @@ static int by_store_ctrl_ex(X509_LOOKUP *ctx, int cmd, const char *argp,
                             long argl, char **retp, OSSL_LIB_CTX *libctx,
                             const char *propq)
 {
+    char *data;
```
[the remark this answers is not in the document]
Comment (JiangJias) [reply crypto/x509/by_store.c]: Thanks, I have already submitted a fixup.

File: crypto/x509/by_store.c [hunk 2]
```diff
@@ -124,7 +124,10 @@ static int by_store_ctrl_ex(X509_LOOKUP *ctx, int cmd, const char *argp,
                 uris = sk_OPENSSL_STRING_new_null();
                 X509_LOOKUP_set_method_data(ctx, uris);
             }
-            return sk_OPENSSL_STRING_push(uris, OPENSSL_strdup(argp)) > 0;
+            char *data = OPENSSL_strdup(argp);
```
Comment (paulidale) [inline crypto/x509/by_store.c:127]: Declarations go before all code.  so `char *data` needs to go up to below line 120.

Comment (JiangJias) [reply crypto/x509/by_store.c]: Thanks, I have submitted a fixup.

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the fix.

**Pull request closed without merging**